src/components/Clock.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class Clock extends React.Component {
+  state = {
+    date: new Date(),
+  }
+
+  timer = setInterval(() => {
+    this.setState({ date: new Date() });
+    const { date } = this.state;
+    // eslint-disable-next-line
+    console.log(date.toLocaleTimeString());
+  }, 1000);
+
+  componentDidMount() {
+    return this.timer;
+  }
+
+  componentDidUpdate(prevProps) {
+    const oldName = prevProps.name;
+    const newName = this.props.name;
+
+    if (oldName !== newName) {
+      // eslint-disable-next-line
+      console.log(`The Clock was renamed from ${oldName} to ${newName}`);
+    }
+  }
```

Comment:
It's great that you have used componentDidUpdate, but for this task you should only console.log the new name only when clicking the button (so you could easily put the 'concole.log to your 'randomName' method)